sda-commons-starter-web/src/test/java/org/sdase/commons/spring/boot/web/security/handler/ValidationExceptionTest.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2022- SDA SE Open Industry Solutions (https://www.sda.se)
+ *
+ * Use of this source code is governed by an MIT-style
+ * license that can be found in the LICENSE file or at
+ * https://opensource.org/licenses/MIT.
+ */
+package org.sdase.commons.spring.boot.web.security.handler;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.jupiter.api.Test;
+import org.sdase.commons.spring.boot.web.security.handler.app.ValidationExceptionBook;
+import org.sdase.commons.spring.boot.web.security.handler.app.ValidationExceptionHandlerTestApp;
+import org.sdase.commons.spring.boot.web.testing.auth.AuthMock;
+import org.sdase.commons.spring.boot.web.testing.auth.EnableSdaAuthMockInitializer;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.server.LocalServerPort;
+import org.springframework.http.HttpStatus;
+import org.springframework.test.context.ContextConfiguration;
+
+@SpringBootTest(
+    classes = ValidationExceptionHandlerTestApp.class,
+    webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@ContextConfiguration(initializers = {EnableSdaAuthMockInitializer.class})
+class ValidationExceptionTest {
+
+  @LocalServerPort private int port;
+
+  @Autowired private AuthMock authMock;
+
+  @Test
+  void testValidation() {
+
+    authMock.authorizeAnyRequest().allow();
+
+    var response =
+        authMock
+            .authentication()
+            .authenticatedClient()
+            .postForEntity(
+                "http://localhost:" + port + "/api/bookSpring",
+                new ValidationExceptionBook("title", "author"),
+                ValidationExceptionBook.class);
+
+    ValidationExceptionBook body = response.getBody();
+    assertThat(body).isNotNull();
+    assertThat(body.title()).isEqualTo("title");
+    assertThat(body.author()).isEqualTo("author");
+  }
+
+  @Test
+  void testValidationSpring() {
+
+    authMock.authorizeAnyRequest().allow();
+
+    var response =
+        authMock
+            .authentication()
+            .authenticatedClient()
+            .postForEntity(
+                "http://localhost:" + port + "/api/bookSpring",
+                new ValidationExceptionBook("title", null),
+                ValidationExceptionBook.class);
+
+    assertThat(response.getStatusCode()).isEqualTo(HttpStatus.UNPROCESSABLE_ENTITY);
+  }
+
+  @Test
+  void testValidationJakarta() {
+
+    authMock.authorizeAnyRequest().allow();
+
+    var response =
+        authMock
+            .authentication()
+            .authenticatedClient()
+            .postForEntity(
+                "http://localhost:" + port + "/api/bookJakarta",
+                new ValidationExceptionBook("title", null),
+                String.class);
```

Comment:
Is this really the case you want to test? Since the controller method adds `@NotNull` to the RequestBody-object, I expected that you want to post `null`.

Reply:
This is what i wanted to test because it is throwing the `HandlerMethodValidationException`

However i added a test for ResponseBody null
Turns out this is causing another Exception `HttpMessageNotReadableException`
I extended our ValidationHandler again to include this execption

Although i think this is the intended behaviour i have concerns that this could break test for users of our library.
I added a property to allow to disable it `security.validation-exception-handler-enabled=false`